src/Common/src/Microsoft/Win32/SafeHandles/SafeBioHandle.Unix.cs

```diff
@@ -5,27 +5,53 @@
 using System;
 using System.Security;
 using System.Runtime.InteropServices;
+using System.Diagnostics;
 
 namespace Microsoft.Win32.SafeHandles
 {
     [SecurityCritical]
     internal sealed class SafeBioHandle : SafeHandle
     {
+        private SafeHandle _parent;
+
         private SafeBioHandle() :
             base(IntPtr.Zero, ownsHandle: true)
         {
         }
 
         protected override bool ReleaseHandle()
         {
-            Interop.Crypto.BioDestroy(handle);
+            if (_parent != null)
+            {
+                _parent.DangerousRelease();
+                _parent = null;
+            }
+            else
+            {
+                Interop.Crypto.BioDestroy(handle);
+            }
             SetHandle(IntPtr.Zero);
             return true;
         }
 
         public override bool IsInvalid
         {
-            get { return handle == IntPtr.Zero; }
+            get
+            {
+                // If handle is 0, we're invalid.
+                // If we have a _parent and they're invalid, we're invalid.
+                return handle == IntPtr.Zero || (_parent != null && _parent.IsInvalid);
+            }
+        }
+
+        internal void TransferOwnershipToParent(SafeHandle parent)
+        {
+            Debug.Assert(_parent == null, "Expected no existing parent");
```

Comment:
Should you also assert that `handle` is non-zero?  We seem to assume that elsewhere.